libnmstate/nm/iplib.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import ipaddress
+import six
+
+
+IPV6_LINK_LOCAL_NETWORK = "fe80::/10"
+
+
+def _is_subnet_of(a, b):
+    """
+    Copy from cpython 3.7 Lib/ipaddress.py file which is licensed under PSF,
+    copyright 2007 Google Inc.
+    """
+    try:
+        # Always false if one is v4 and the other is v6.
+        if a._version != b._version:
+            raise TypeError('{a} and {b} are not of the same version')
+        return (b.network_address <= a.network_address and
+                b.broadcast_address >= a.broadcast_address)
+    except AttributeError:
+        raise TypeError('Unable to test subnet containment '
+                        'between {a} and {b}')
+
+
+def is_subnet_of(network_a, network_b):
+    """
+    Check whether network_a is subnet of network_b.
+    """
+    a = ipaddress.ip_network(six.u(str(network_a)), strict=False)
+    b = ipaddress.ip_network(six.u(str(network_b)), strict=False)
+    if hasattr(a, 'subnet_of'):
```

Comment:
Even if I create a psudo `ipaddess` module, I still need another module to provide a function for
subnet calculation, then why two module?

Can you elaborate on the `compat` module? Is that [this module][1]?

[1]: https://python-utilities.readthedocs.io/en/latest/compat.html

Reply:
Ahh sorry, I forgot to give an example: https://github.com/oVirt/vdsm/blob/master/lib/vdsm/common/compat.py

You basically need to inherit in case it is not included and add what you need.
At the place it is used, it should not know you did it.